drivers/hmis/spec/system/hmis/ac_hmis/ac_ce_referral_workflows_spec.rb

```diff
@@ -0,0 +1,411 @@
+          expect(page).not_to have_content('Enrollment Link') # Can't click into the enrollment
+        end
+
+        # Provider denies the referral
+        complete_ce_step('Provider Outcome') do
+          fill_in_step_with_notes(notes: 'Provider declines')
+          mui_radio_choose 'Decline - Submit Referral for Denial Review', from: 'Decision'
+          mui_radio_choose 'Inability to complete intake', from: 'Decline Reason'
+        end
+
+        # Denial Review task is available, but the provider user can't open it
+        expect(page).to have_content('Denial Review Available Today')
+        expect(page).not_to have_button('Start step: Denial Review')
+      end
+
+      complete_ce_step('Denial Review') do
+        fill_in_step_with_notes(notes: 'No, send back!')
+        mui_radio_choose 'Send Back', from: 'Decision'
+      end
+
+      expect(page).to have_content('Provider Outcome (Second Attempt)')
+
+      with_user_impersonated(provider.id) do
+        expect(page).to have_content('Provider Outcome (Second Attempt) Available Today Assigned to you')
+
+        complete_ce_step('Provider Outcome (Second Attempt)') do
+          fill_in_step_with_notes(notes: 'Provider accepts this time')
+          mui_radio_choose 'Accept - Add to Project', from: 'Decision'
+          expect(page).to have_content('The client will be added to the project as Incomplete.')
+        end
+
+        expect(page).to have_content('Confirm Success Available Today')
+      end
+
+      complete_ce_staff_confirm_success_step('Alice A', unit)
+    end
+
+    it 'completes the change provider outcome => denied path' do
+      ce_staff_complete_initial_steps
+
+      # Impersonate Paul Provider and verify they can see the referral
+      with_user_impersonated(provider.id) do
+        confirm_provider_functionality(
+          client_name: 'Alice A',
+          previous_step_name: 'Initial Review',
+          previous_step_content: 'Referral for Alice A is in progress',
+          previous_note_text: 'Hello Paul, this referral is in your court now',
+        )
+
+        # Provider denies the referral
+        complete_ce_step('Provider Outcome') do
+          fill_in_step_with_notes(notes: 'Provider approves')
+          mui_radio_choose 'Accept - Add to Project', from: 'Decision'
+          expect(page).to have_content('The client will be added to the project as Incomplete.')
+        end
+
+        # Submit the Change Provider Outcome step to move the referral to denied pending status
+        complete_ce_step('Change Provider Outcome (Optional)') do
+          fill_in_step_with_notes(notes: 'Changing the result')
+          mui_radio_choose 'Inability to complete intake', from: 'Decline Reason'
+          expect(page).to have_content('The client\'s in-progress enrollment in this project will be deleted.')
+        end
+
+        expect(page).to have_content('Denial Review Available Today')
+        expect(page).to have_content('Denial Pending')
```

Comment:
This is OK but I think we might as well check that the referral status was updated on the record itself, like
`expect(referral.custom_status.name).to eq('Denial Pending')`